tests/test_visibility.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 New Vector Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Optional
+
+import aiounittest
+
+from room_access_rules import ACCESS_RULES_TYPE, AccessRules
+from tests import create_module, PUBLIC_ROOM_ID, new_access_rules_event
+
+
+class RoomVisibilityTestCase(aiounittest.AsyncTestCase):
+    def setUp(self) -> None:
+        self.module = create_module()
+        self.user_id = "@someuser:example.com"
+
+    async def test_change_rules_when_published(self):
+        """Tests that the rule of a room cannot be set to something other than
+        'restricted' when the room is published in the room directory.
+        """
+        state = {
+            (ACCESS_RULES_TYPE, ""): new_access_rules_event(
+                self.user_id, PUBLIC_ROOM_ID, AccessRules.RESTRICTED,
+            )
+        }
+
+        # Check that we can't change the rule to 'unrestricted'.
+        allowed, _ = await self.module.check_event_allowed(
```

Comment:
Which function are we calling here---this one?

https://github.com/vector-im/synapse-room-access-rules/blob/084ba5b6c986680dfab117b6bc869f8377b0b508/room_access_rules/__init__.py#L335-L339

The return type annotation for that says it returns either a `bool` or a `dict`. We seem to be expecting a Tuple here of two things and discarding the second, so I don't understand what's going on here.

Reply:
Oh sorry, this depends on #3. Let me take a look at that now.

Reply:
Yeah this callback wasn't ported correctly and had the old return type/values, I fixed it in #3